### PR TITLE
Add an `exportBuffer` action

### PR DESCRIPTION
**Pull request by zadjii-msft**
This adds an action for the context menu entry we added in #11062. That PR added support for exporting the buffer, exclusively through the tab item's context menu. This adds an action that can additionally be bound, which also can export the buffer to a file. This action accepts a `path` param. If empty/ommitted, then the Terminal will prompt for the file to export the buffer to. 

* Does a part of #9700
* Spec in #11090, but I doubt this is contentious
* [x] This will satisfy #12052
* [x] I work here
* [x] docs added: https://github.com/MicrosoftDocs/terminal/pull/479

### CODE REVIEW

**Review by carlos-zamora — APPROVED** [review]
Looks good to me! Thanks!

**Comment by carlos-zamora** [comment]
oh, the docs. Update them plz haha

File: src/cascadia/TerminalApp/TabManagement.cpp [hunk 1]
```diff
@@ -362,25 +364,38 @@ namespace winrt::TerminalApp::implementation
...
+                        THROW_IF_FAILED(dialog->SetFileTypeIndex(1)); // the array is 1-indexed
+                        THROW_IF_FAILED(dialog->SetDefaultExtension(L"txt"));
+
+                        // Default to using the tab title as the file name
+                        THROW_IF_FAILED(dialog->SetFileName((control.Title() + L".txt").c_str()));
+                    });
+                }
+                else
+                {
+                    // The file picker isn't going to give us paths with
```
Comment (miniksa) [inline src/cascadia/TerminalApp/TabManagement.cpp:393]: Honestly wouldn't hurt you to call this on either path just in case, but it's fine.

**Review by miniksa — APPROVED** [review]
Man it's become a lot easier to do things around here with X macros hasn't it?

**Pull request merged**